### PR TITLE
chore: refactor the repo into a monorepo managed by lerna

### PR DESCRIPTION
**Pull request by raymondfeng**
Signed-off-by: Raymond Feng <enjoyjava@gmail.com>

### CODE REVIEW

File: packages/recommender/src/__tests__/integration/recommender.integration.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
```
Comment (b-admike) [inline packages/recommender/src/__tests__/integration/recommender.integration.ts]: These test files are just moved files and not new additions, right? Are there new things you added besides refactoring imports to make it work with the new project structure?

File: lerna.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      "concurrency": 8
+    },
+    "version": {
+      "conventionalCommits": true,
+      "message": "chore: publish release"
+    },
+    "publish": {
+    },
+    "run": {
+      "prefix": false,
```
[the remark this answers is not in the document]
Comment (raymondfeng) [reply lerna.json]: Yes.

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "loopback4-example-shopping",
+  "name": "loopback4-example-shopping-monorepo",
```
[the remark this answers is not in the document]
Comment (raymondfeng) [reply package.json]: The top level package is just for the repo.

File: package.json [hunk 4]
```diff
@@ -41,53 +40,22 @@
   "repository": {
     "type": "git"
   },
+  "private": true,
```
[the remark this answers is not in the document]
Comment (raymondfeng) [reply package.json]: The `private` flag is only for the root package.

File: packages/shopping/tsconfig.build.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "$schema": "http://json.schemastore.org/tsconfig",
+  "extends": "@loopback/build/config/tsconfig.common.json",
+  "compilerOptions": {
+    "rootDir": "src"
+  },
+  "include": ["src", "src/**/*.json"]
```
[the remark this answers is not in the document]
Comment (raymondfeng) [reply packages/shopping/tsconfig.build.json]: We need to include json files are there are json definitions for datasources.

File: packages/recommender/src/__tests__/integration/recommender.integration.ts (hunk 1 above)
Comment (raymondfeng) [reply packages/recommender/src/__tests__/integration/recommender.integration.ts]: This one is new.

File: packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts [hunk 6]
```diff
@@ -6,18 +6,17 @@
...
-import {Server} from 'http';
-import * as _ from 'lodash';
-import {JWTAuthenticationService} from '../../src/services/JWT.authentication.service';
 import {
-  PasswordHasherBindings,
-  JWTAuthenticationBindings,
-} from '../../src/keys';
-const recommendations = require('../../recommender/recommendations.json');
+  createRecommendationServer,
+  HttpServer,
```
Comment (b-admike) [inline packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts]: Is this a re-export from `@loopback/http-server`?

File: packages/recommender/src/__tests__/integration/recommender.integration.ts (hunk 1 above)
Comment (b-admike) [reply packages/recommender/src/__tests__/integration/recommender.integration.ts]: Nice 👍

File: packages/recommender/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: loopback4-example-recommender
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+export * from './mock-recommendation-app';
+export * from '@loopback/http-server';
```
Comment (b-admike) [inline packages/recommender/src/index.ts]: 👍

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (b-admike) [reply package.json]: I see, in that case we can leave it there.

**Pull request merged**